libconway/src/grids.rs

```diff
@@ -224,14 +224,50 @@ impl BitGrid {
         }
     }
 
-    /// Rotates pattern in the specified direction around the specified center point.
+    /// Rotates pattern with top-left corner at `(0,0)` in the grid and lower right corner at
+    /// `(width - 1, height - 1)` in the specified direction. This may change the dimensions of the
+    /// grid.
     ///
     /// # Errors
     ///
-    pub fn rotate(&mut self, rotation: Rotation, (center_col, center_row): (usize, usize)) -> Result<(), Box<dyn Error>> {
+    /// An error is returned if the width or height are out of range.
+    pub fn rotate(&mut self, width: usize, height: usize, rotation: Rotation) -> Result<(), Box<dyn Error>> {
+        if width > self.width() || height > self.height() {
+            return Err(format!("Expected passed-in width={} and height={} to be less than grid width={} and height={}",
+                    width, height, self.width(), self.height()).into());
+        }
         let new_width_in_words = (self.height() - 1)/64 + 1;   // number of words needed for this many cells
         let new_height = self.width();
         let mut new = BitGrid::new(new_width_in_words, new_height);
+        for row in 0 .. height {
+            let new_col = match rotation {
+                Rotation::CCW => row,
+                Rotation::CW => height - row - 1,
+            };
+            let new_col_idx = new_col/64; // the column index in the new grid
+            let mut col = 0;
+            'rowloop:
+            for col_idx in 0 .. self.width_in_words() {
```

Comment:
I think the `'rowloop:` belongs on the same line as the for statement if we're following rustfmt conventions. Do correct me if I am wrong about this. I'm basing it off of `'label: loop {}`